rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -67,10 +69,19 @@ protected TbCreateAlarmNodeConfiguration loadAlarmNodeConfig(TbNodeConfiguration
     @Override
     protected ListenableFuture<TbAlarmResult> processAlarm(TbContext ctx, TbMsg msg) {
         String alarmType;
+        String alarmSeverity;
         final Alarm msgAlarm;
 
         if (!config.isUseMessageAlarmData()) {
             alarmType = TbNodeUtils.processPattern(this.config.getAlarmType(), msg);
+            if(this.config.isDynamicSeverity()) {
+                alarmSeverity = TbNodeUtils.processPattern(this.config.getSeverity(), msg);
+                if(!EnumUtils.isValidEnum(AlarmSeverity.class, alarmSeverity)) {
+                    ctx.tellFailure(msg, new TbNodeException("Used incorrect pattern or Alarm Severity not included in message: " + alarmSeverity));
+                    return null;
+                }
+                this.config.setSeverity(alarmSeverity);
```

Comment:
You are overwriting the template with the calculated value.

Reply:
@ashvayka Refactored